### PR TITLE
fix "Expected all tensors to be on the same device, but found at least two devices" error

### PR DESCRIPTION
**Pull request by yao-matrix**

### CODE REVIEW

**Comment by yao-matrix** [comment]
@SunMarc , could you pls help review? Thx very much.

File: src/diffusers/models/unets/unet_2d_blocks.py [hunk 1]
```diff
@@ -2557,7 +2557,8 @@ def forward(
                     b1=self.b1,
                     b2=self.b2,
                 )
-
+            if hidden_states.device != res_hidden_states.device:
+                res_hidden_states = res_hidden_states.to(hidden_states.device)
```
Comment (SunMarc) [inline src/diffusers/models/unets/unet_2d_blocks.py]: we shouldn't need that since both hidden_states and res_hidden_states should be on the same device no ? The pre-forward hook added by accelerate should be move all the inputs to the same device.

Comment (yao-matrix) [reply src/diffusers/models/unets/unet_2d_blocks.py]: @SunMarc , i suppose this is a corner case? `torch.cat` is a weight-less function, so seems cannot covered by the pre-forward hook set by `accelerate`...

Comment (SunMarc) [reply src/diffusers/models/unets/unet_2d_blocks.py]: I mean since `hidden_states` and `res_hidden_states_tuple` are in the forward definition, they should be moved to the same device by the pre-forward hook added by accelerate

Comment (yao-matrix) [reply src/diffusers/models/unets/unet_2d_blocks.py]: @SunMarc We run into a corner case here. Since we have 8 cards here, so the determined `device_map`(by https://github.com/huggingface/diffusers/blob/1bc6f3dc0f21779480db70a4928d14282c0198ed/src/diffusers/models/model_loading_utils.py#L64C5-L64C26) is 

> device_map: OrderedDict([('conv_in', 0), ('time_proj', 0), ('time_embedding', 0), ('down_blocks.0', 0), ('down_blocks.1.resnets.0', 1), ('up_blocks.0.resnets.0', 1), ('up_blocks.0.resnets.1', 2), ('up_blocks.0.upsamplers', 2), ('up_blocks.1', 3), ('mid_block.attentions', 3), ('conv_norm_out', 4), ('conv_act', 4), ('conv_out', 4), ('mid_block.resnets', 4)])

We can see `UpBlock` is not the atomic module, its submodules are assigned to different devices(up_blocks.0.resnets.0, up_blocks.0.resnets.1), so pre-hook for `UpBlock` will not help in this case. And since `torch.cat` is not pre-hooked(and cannot since it's a function rather than a module?), so the issue happens.   

If there is no a `torch.cat` btw the sub-blocks in `UpBlock`, things will be all fine.

Comment (yao-matrix) [reply src/diffusers/models/unets/unet_2d_blocks.py]: @SunMarc, need your inputs in how to proceed for this corner case, thx.

Comment (yao-matrix) [reply src/diffusers/models/unets/unet_2d_blocks.py]: @SunMarc  We can see a similar case in `transformers` ut `pytest -rA tests/models/chameleon/test_modeling_chameleon.py::ChameleonVision2SeqModelTest::test_model_parallel_beam_search` w/ 2 cards, the error log is "RuntimeError: Expected all tensors to be on the same device, but found at least two devices,src/transformers/models/chameleon/modeling_chameleon.py", the reason is even `residual` is in the same device as `hidden_states` at the beginning, but after they went through some operators as both input and output, they finally placed to different device, but when they come to `+` which is not a nn.Module(so accelerate cannot pre-hook it), error happens.  Do you have some insights on such issues?

Comment (yao-matrix) [reply src/diffusers/models/unets/unet_2d_blocks.py]: @SunMarc, could you share your insights on the issue i mentioned above? thx very much.

Comment (SunMarc) [reply src/diffusers/models/unets/unet_2d_blocks.py]: Sorry for the long wait @yao-matrix , if you add `UpBlock2D` in `_no_split_modules` of `UNet2DConditionModel`, the test should pass ! 
`_no_split_modules = ["BasicTransformerBlock", "ResnetBlock2D", "CrossAttnUpBlock2D", "UpBlock2D"]`
I've tested on my end and it works. 

As for ChameleonVision2SeqModelTest, we probably need to also update _no_split_modules.

**Comment by yao-matrix** [comment]
@SunMarc , done thx, pls help review again.

**Review by SunMarc — APPROVED** [review]
LGTM !

File: tests/models/test_modeling_common.py [hunk 2]
```diff
@@ -1945,7 +1944,7 @@ def test_push_to_hub_library_name(self):
         delete_repo(self.repo_id, token=TOKEN)
 
 
-@require_torch_gpu
+@require_torch_accelerator
 @require_torch_2
```
Comment (sayakpaul) [inline tests/models/test_modeling_common.py:1948]: This change is unrelated to this PR. Going forward prefer not including unrelated changes in a particular PR.

**Review by sayakpaul — APPROVED** [review]
Thanks!

File: tests/models/test_modeling_common.py (hunk 2 above)
Comment (yao-matrix) [reply tests/models/test_modeling_common.py:1948]: sure, will follow the rule going forward.

File: tests/models/unets/test_models_unet_2d_condition.py [hunk 3]
```diff
@@ -358,7 +358,7 @@ class UNet2DConditionModelTests(ModelTesterMixin, UNetTesterMixin, unittest.Test
     model_class = UNet2DConditionModel
     main_input_name = "sample"
     # We override the items here because the unet under consideration is small.
-    model_split_percents = [0.5, 0.3, 0.4]
+    model_split_percents = [0.5, 0.34, 0.4]
```
Comment (yao-matrix) [inline tests/models/unets/test_models_unet_2d_condition.py:362]: since `UpBlock2D` is not split now, need update 0.3->0,34 split percent to make UT pass

**Comment by yao-matrix** [comment]
@sayakpaul , pls help retrigger the ci, thx very much.
